### ISSUE TITLE
My Vault City has disappeared off of the map!

### ISSUE THREAD

**Issue by Squabbywaboose** [alexbatalov/fallout2-ce#397]
So I have been enjoying Fallout 2 on the iphone with no glaring issues. I already spent sometime in Vault City and have gotten my Citizenship there so it's not like I haven't been there already. And when I click the "vault city" button when traveling, it just brings me to the right of Gecko like my character is trying to go off screen to get to it (which is funny because isn't Vault City to the left of Gecko?) so I think I'm done with Vault City. Anyone else have a similar issue? The only time I opened the configurations back up was when I had to get the music working again, maybe I messed something up. Either way, Fuck Vault City!

**Comment by manbehindthemadness** [comment]
I am also noticing this on MacOs, the shift keys don't seem to work at all.

**Comment by manbehindthemadness** [comment]
Update: I actually retried this on a Ubuntu machine using Wine and the shift to hi light function didn't work there either. I am wondering if this is the base version of Fallout2 used, or a shared parameter of some kind...

**Comment by xparaly** [comment]
I'm on windows and my shift key does not highlight items either.  If I launch Fallout2.exe, the shift key works as expected.

**Comment by manbehindthemadness** [comment]
After some experimentation I have discovered that this seems to be an issue loading the restoration project additions, I created a fully functioning install from the original cd on Ubuntu using wine, applied the restoration project, tested to confirm it was working, (had to install vulkan and open gl mesa, in addition to adding the draw exclusion environment variable). I then copied this directory to a MacBook and performed the fallout2-ce setup procedures. This resulted in an application crash when loading into the temple environment just after the white-screen transition. This could only be resolved by removing the gl_ammomod.int file from the scripts directory. On the second run the mouse stoped working in full-screen mode; however, the application didn't crash. Unfortunately the "magic rock" didn't appear in the temple environment indicating the modifications loaded successfully.

This was performed with the latest release binaries from both projects.

To summarize the above: there isn't a problem with the key-mapper, it's a problem loading the after market patches and mods, the shift-to-hilight functionality doesn't exist in the Fallout2 version 1.02 base image.

**Comment by manbehindthemadness** [comment]
Just to confirm that there wasn't some odd platform issue, I compiled the project from source. The issue remains. I think this will resolve itself as the sfall portions are added into the application support; however, we are out of luck until that happens.

**Comment by phobos2077** [comment]
Probably rename the issue. If it's really about missing keypress hook and not platform related.

**Comment by manbehindthemadness** [comment]
> Maybe rename the issue. If it's really about missing keypress hook and not platform related.

What I am seeing is not a faulty keypress, this is a misleading take on my part: If I play the vanilla game v1.02, the shift key doesn't work regardless (al least not to outline items {this explains my observation in relation to the base image of the game, mine coming from the original CD}). If I install the Fallout Restoration Project (again using wine) The ship-to-outline functions work as expected in addition to the magic version rock in the temple environment. If I use this file structure as a base for fallout2-ce the game fails to load into the environments until the gl_ammomod.int script is removed; furthermore, when the game does indeed launch, the version rock is not present. This indicated to me that it's not a problem with the shift key, rather it's a problem with the original base image not using the shift key as seen in later distros in addition to the mods within the Fallout Restoration Project not being evaluated resulting in the observed behavior.

**Comment by eglische** [comment]
I'm under windows here, followed the installation instruction for fallout sonora 1.14 english translation:
https://www.moddb.com/mods/fallout-sonora-english/downloads/fallout-sonora-114-sonora-dayglow-114-english-translation-patch

shift key (dx key 42) also not working, 
-tried us layout -> not working
-tried older sonora translation for v. 1.11 -> not working
-made sure this setting is set in the ddraw.ini: ToggleItemHighlightsKey=42

the game files comes from a us 1.02 GOG fo2 installation

**Comment by phobos2077** [comment]
Why are you editing one issue into completely different one? This confuses everyone.

**Comment by manbehindthemadness** [comment]
> Why are you editing one issue into completely different one? This confuses everyone.

Incredibly bad practice. Editing a populated trouble report into a completely different issue just to close it out. Like WTH?
